laplace/utils/feature_extractor.py

```diff
@@ -66,6 +91,20 @@ def forward_with_features(self, x: torch.Tensor) -> Tuple[torch.Tensor, torch.Te
         """
         out = self.forward(x)
         features = self._features[self._last_layer_name]
+
+        if len(features.shape) > 2 and self.feature_reduction is not None:
+            n_intermediate_dims = len(features.shape) - 2
+
+            if self.feature_reduction == FeatureReduction.PICK_FIRST:
+                features = features[:, *([0] * n_intermediate_dims), :].squeeze()
+            elif self.feature_reduction == FeatureReduction.PICK_LAST:
+                features = features[:, *([0] * n_intermediate_dims), :].squeeze()
+            else:
```

Comment:
Maybe check explicitly `if self.feature_reduction == FeatureReduction.AVERAGE` and add an else-statement that raises a `ValueError`?